lib/packwerk/deprecated_references.rb

```diff
@@ -47,6 +47,20 @@ def add_entries(reference, violation_type)
       @new_entries[reference.constant.package.name] = package_violations
     end
 
+    def stale_violations?
+      deprecated_references.each_key do |package|
```

Comment:
This whole structure…

```ruby
deprecated_references.each_key do |package|
  deprecated_references[package].each_key do |constant_name|
    # …
    stale_violations = … deprecated_references.dig(package, constant_name, "files") …
    # …
    if stale_violations.present?
      return true
    end
  end
end
false
```

…could instead be written as:

```ruby
deprecated_references.any? do |package, package_violations|
  package_violations.any? do |constant_name, entries_for_file|
    # …
    stale_violations = … entries_for_file["files"] …
    # …
    stale_violations.present?
  end
end
```

The changes are:

1. use `Enumerable#any?` to check as many elements as necessary and decide whether to return `true` or `false` overall, rather than using `Hash#each_key` and breaking out of the loop explicitly ourselves;
2. use the values as well as the keys from the outer `deprecated_references` hash as we iterate over it, rather than throwing away the values and then using the keys to look them up again;
3. likewise, use the values as well as the keys from the inner `package_violations` hash as we iterate over it rather than `dig`ing into the outer `deprecated_references`; and
4. implicitly returning `stale_violations.present?` from the inner block (for use by `#any?`) rather than breaking out with an explicit `return`.

I think the second version makes it easier to see at a glance that we’re iterating over every violation in `deprecated_references` (two levels deep) and checking whether `any?` of them are stale. It also happens to be a little bit more concise which is a nice bonus. What do you think?